### PR TITLE
Llama tutorial update related to 24.04 release

### PR DESCRIPTION
**Pull request by oandreeva-nv**
I've updated Llama2 TRT-LLM tutorial with instructions for how to build an engine inside the Triton container (started 24.04).

I've also added a bash script, which includes all steps, mentioned in this tutorial.

### CODE REVIEW

File: Popular_Models_Guide/Llama2/trtllm_guide.md [hunk 1]
```diff
@@ -55,102 +68,140 @@ git lfs install
...
 
-Alternatively, you can follow instructions [here](https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/README.md) to build Triton Server with Tensorrt-LLM Backend if you want to build a specialized container.
+Alternatively, you can follow instructions
+[here](https://github.com/triton-inference-server/tensorrtllm_backend?tab=readme-ov-file#build-the-docker-container)
+to build Triton Server with Tensorrt-LLM Backend if you want
+to build a specialized container.
 
 Don't forget to allow gpu usage when you launch the container.
 
+> Optional: For simplicity, we've condenced all following steps into
```
Comment (jbkyang-nvi) [inline Popular_Models_Guide/Llama2/trtllm_guide.md:96]:
```suggestion
> Optional: For simplicity, we've condensed all following steps into
```

File: Popular_Models_Guide/Llama2/trtllm_guide.md [hunk 2]
```diff
@@ -28,23 +28,36 @@
...
 TensorRT-LLM is Nvidia's recommended solution of running Large Language
 Models(LLMs) on Nvidia GPUs. Read more about TensoRT-LLM [here](https://github.com/NVIDIA/TensorRT-LLM)
-and Triton's TensorRTLLM Backend [here](https://github.com/triton-inference-server/tensorrtllm_backend).
+and Triton's TensorRT-LLM Backend [here](https://github.com/triton-inference-server/tensorrtllm_backend).
 
 *NOTE:* If some parts of this tutorial doesn't work, it is possible that there
-are some version mismatches between the `tutorials` and `tensorrt_backend` repository.
-Refer to [llama.md](https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/docs/llama.md)
-for more detailed modifications if necessary. And if you are familier with python, you can also try using [High-level API](https://github.com/NVIDIA/TensorRT-LLM/blob/main/examples/high-level-api/README.md) for LLM workflow.
+are some version mismatches between the `tutorials` and `tensorrt_backend`
```
Comment (krishung5) [inline Popular_Models_Guide/Llama2/trtllm_guide.md:34]: `tensorrt_backend` -> `tensorrtllm_backend`?

File: Popular_Models_Guide/Llama2/trtllm_guide.md [hunk 3]
```diff
@@ -55,102 +68,140 @@ git lfs install
...
-    >
-    >   ```bash
-    >    python3 /tensorrtllm_backend/tensorrt_llm/examples/llama/run.py --engine_dir=/engines/1-gpu/ --max_output_len 100 --tokenizer_dir /Llama-2-7b-hf --input_text "How do I count to ten in French?"
-    >    ```
+
+TensorRT-LLM requires each model to be compiled for the configuration
+you need before running. To do so, before you run your model for the first time
+on Triton Server you will need to create a TensorRT-LLM engine.
+
+Starting with 24.04 release, Triton Server TensrRT LLM container comes with
```
Comment (krishung5) [inline Popular_Models_Guide/Llama2/trtllm_guide.md:114]:
```suggestion
Starting with 24.04 release, Triton Server TensorRT-LLM container comes with
```

Comment (jbkyang-nvi) [inline Popular_Models_Guide/Llama2/trtllm_guide.md]: Link here?

File: Popular_Models_Guide/Llama2/trtllm_guide.md (hunk 2 above)
Comment (oandreeva-nv) [reply Popular_Models_Guide/Llama2/trtllm_guide.md]: updated

File: Popular_Models_Guide/Llama2/trtllm_guide.md (hunk 3 above)
Comment (oandreeva-nv) [reply Popular_Models_Guide/Llama2/trtllm_guide.md]: updated

File: Popular_Models_Guide/Llama2/trtllm_guide.md [hunk 4]
```diff
@@ -26,25 +26,136 @@
...
+
+### Prepare Triton model repository
+Triton CLI has a single command `triton import` that automatically converts HF
+checkpoint into TensorRT-LLM checkpoint format, builds TensorRT-LLM engines,
+and prepares a Triton model repository:
+```bash
+ENGINE_DEST_PATH=/engines triton import -m llama-2-7b --backend tensorrtllm
+```
+
+Please, note that specifying `ENGINE_DEST_PATH` is optional, but recmmended
```
Comment (krishung5) [inline Popular_Models_Guide/Llama2/trtllm_guide.md:94]:
```suggestion
Please, note that specifying `ENGINE_DEST_PATH` is optional, but recommended
```

File: Popular_Models_Guide/Llama2/trtllm_guide.md [hunk 5]
```diff
@@ -26,25 +26,136 @@
...
+```
+triton start
+```
+
+### Send an inference request
+Use the [generate endpoint](https://github.com/triton-inference-server/tensorrtllm_backend/tree/release/0.5.0#query-the-server-with-the-triton-generate-endpoint).
+to send an inference request to the deployed model.
+
+```bash
+curl -X POST localhost:8000/v2/models/ensemble/generate -d '{"text_input": "What is ML?", "max_tokens": 50, "bad_words": "", "stop_words": "", "pad_id": 2, "end_id": 2}'
```
Comment (krishung5) [inline Popular_Models_Guide/Llama2/trtllm_guide.md:136]:
```suggestion
curl -X POST localhost:8000/v2/models/llama-2-7b/generate -d '{"text_input": "What is ML?", "max_tokens": 50, "bad_words": "", "stop_words": "", "pad_id": 2, "end_id": 2}'
```

**Review by krishung5 — APPROVED** [review]
LGTM!

**Pull request merged**